### ISSUE TITLE
Update build system to fail more quickly

### ISSUE THREAD

**Issue by sobolevnrm** [Electrostatics/pdb2pqr#424]
Update GitHub workflows to fail quickly on linting errors before running the more time-consuming functional tests.

**Comment by kiyoon** [comment]
Or, you can separate the test to a different workflow to identify which part is failing more clearly

**Comment by sobolevnrm** [comment]
@kiyoon can you please explain the differences between these three instances of the ruff linting checks?

1. https://github.com/Electrostatics/pdb2pqr/blob/9dd79cb9589ea8fac701dc40a70698baa7dd009e/.github/workflows/python-package.yml#L30
2. https://github.com/Electrostatics/pdb2pqr/blob/9dd79cb9589ea8fac701dc40a70698baa7dd009e/.github/workflows/check-styles.yml#L12
3. https://github.com/Electrostatics/pdb2pqr/blob/9dd79cb9589ea8fac701dc40a70698baa7dd009e/.github/workflows/check-styles.yml#L37

**Comment by kiyoon** [comment]
1. It's a syntax error check. It already existed in the CI with flake8, so I kept the same rules.
2. You can fix by `ruff format .` (apply ruff formatter)
3. You can fix by `ruff check --select I --fix .` (fix ruff import sorting lint)

So 2, 3 are not as important as 1.

**Comment by kiyoon** [comment]
If you go in the action, there is a summary of the run where you can see the diff output of how you would fix them. Currecntly they all pass so there's no output here

<img width="1055" alt="image" src="https://github.com/user-attachments/assets/24467f8b-6a26-4195-8187-7b52978b9b7b" />

**Comment by sobolevnrm** [comment]
Yes, I understand how to view the output.  I was just asking why there were 3 different linting tests.  It seems like the first test duplicates the other two.

**Comment by kiyoon** [comment]
The first test is completely different from the other two, but if you wish you can combine all of them in one. The problem is that there are important tests and there are just style checks. If too many things are combined, maybe it's hard to review others code and ask them to fix stuff, for that I preferred separated tests
